src/client/results/mostfrequent/words/WordItemsList.js

```diff
@@ -1,40 +1,90 @@
 import React, { Component } from 'react';
-import { connect } from 'react-redux';
 
 class WordItemsList extends Component {
   componentDidMount() {
-    $('.ui.checkbox').checkbox();
-    $('.ui.accordion').accordion();
+    if (this.props.isWordsToggledActionHide) {
+      $('.ui.checkbox.words').checkbox('check');
+    } else {
+      $('.ui.checkbox.words').checkbox('uncheck');
+    }
+    $('.ui.accordion.words').accordion({
+      selector: {
+        trigger: '.title .conflated-word-selector',
+      },
+    });
   }
 
-  componentDidUpdate() {
-    $('.ui.checkbox').checkbox();
-    $('.ui.accordion').accordion();
+  componentDidUpdate(nextProps) {
+    if (nextProps.isWordsToggledActionHide !== this.props.isWordsToggledActionHide) {
+      if (this.props.isWordsToggledActionHide) {
+        $('.ui.checkbox.words').checkbox('check');
+      } else {
+        $('.ui.checkbox.words').checkbox('uncheck');
+      }
+    }
   }
 
   render() {
     return (
       <div style={{ height: '300px', overflowY: 'scroll' }}>
-        <div className="ui fluid accordion" style={{ overflow: 'hidden' }}>
-          { this.props.words.map((wordInfo, id) =>
-            <ConflatedWordItem key={wordInfo.word} conflatedWordInfo={wordInfo} toggleMostUsedWord={this.props.toggleMostUsedWord} />) }
+        <div className="ui fluid accordion words" style={{ overflow: 'hidden' }}>
+          {this.props.words.map((wordInfo, id) =>
+            <ConflatedWordItem key={wordInfo.word} accordianIndex={id}
+              conflatedWordInfo={wordInfo}
+              toggleMostUsedWord={this.props.toggleMostUsedWord}
+            />
+          )}
         </div>
       </div>
     );
   }
 }
+WordItemsList.propTypes = {
+  words: React.PropTypes.array,
+  isWordsToggledActionHide: React.PropTypes.bool,
+};
 
-const ConflatedWordItem = ({ toggleMostUsedWord, conflatedWordInfo }) => (
+const ConflatedWordItem = ({ accordianIndex, toggleMostUsedWord, conflatedWordInfo }) => (
   <div>
     <div className="title">
       <div className="ui grid">
-        <div className="two wide right aligned column">
-          <div className="statistic">
-            {conflatedWordInfo.count}
+        <div className="twelve wide column conflated-word-selector">
+          <div className="ui twelve column grid">
+            <div className="two wide right aligned column">
+              <div className="statistic">
+                {conflatedWordInfo.count}
+              </div>
+            </div>
+            <div className="ten wide column">
+              {conflatedWordInfo.word} <i className="dropdown icon" />
+            </div>
           </div>
         </div>
-        <div className="fourteen wide column">
-          {conflatedWordInfo.word} <i className="dropdown icon" />
+        <div className="four wide column">
```

Comment:
Please at Winwardo suggest a cleaner way of doing the below if you think necessary, although this does work :P